src/QueryGrammar.php

```diff
@@ -857,8 +857,22 @@ protected function compileTermsAggregation(array $aggregation): array
             ]
         ];
 
-        if (is_array($aggregation['args']) && isset($aggregation['args']['size'])) {
-            $compiled['terms']['size'] = $aggregation['args']['size'];
+        $allowedArgs = [
```

Comment:
Only a thought, don't have a strong opinion on it, so up to you, but do you think this should be a private property on the class as opposed to a variable?

Reply:
Yeah it's a good thought. I think we might need to do this for more aggregations where in a lot of cases it's just allowing a few specific arguments, which makes me think it might be clearer to keep the allowed args together with the compile function rather than having lots of class variables each of which are only used in a single function.